prediction_market_agent/agents/think_thoroughly_agent/deploy.py

```diff
@@ -17,9 +20,17 @@ def load(self) -> None:
     def answer_binary_market(self, market: AgentMarket) -> Answer | None:
         return self.agent.answer_binary_market(market.question)
 
+    def before(self, market_type: MarketType) -> None:
+        self.agent.update_markets()
+        super().before(market_type=market_type)
+
 
 if __name__ == "__main__":
     agent = DeployableThinkThoroughlyAgent(place_bet=False)
+    # We set the keys manually since it will be required by PMAT
+    if not os.environ.get("OPENAI_API_KEY"):
+        keys = APIKeys()
+        os.environ["OPENAI_API_KEY"] = keys.openai_api_key.get_secret_value()
```

Comment:
Why do you need to set this? Is `OPENAI_API_KEY` used somewhere directly from environment variable instead from APIKeys?

Reply:
Yes, somewhere inside PMAT there is an LLM call that reads it from the .env.

Reply:
https://github.com/gnosis/prediction-market-agent-tooling/blob/main/prediction_market_agent_tooling/tools/is_predictable.py#L47

Reply:
Creating a PR for that now

Reply:
Merged PR